enterprise/cmd/frontend/internal/insights/resolvers/admin_resolver.go

```diff
@@ -114,6 +114,127 @@ func (r *Resolver) InsightViewDebug(ctx context.Context, args graphqlbackend.Ins
 	}
 	return resolver, nil
 }
+func (r *Resolver) RetryInsightSeriesBackfill(ctx context.Context, args *graphqlbackend.BackfillArgs) (*graphqlbackend.BackfillQueueItemResolver, error) {
+	actr := actor.FromContext(ctx)
+	if err := auth.CheckUserIsSiteAdmin(ctx, r.postgresDB, actr.UID); err != nil {
+		return nil, err
+	}
+	var backfillID int
+	err := relay.UnmarshalSpec(args.Id, &backfillID)
+	if err != nil {
+		return nil, errors.Wrap(err, "error unmarshalling the backfill id")
+	}
+	backfillStore := scheduler.NewBackfillStore(r.insightsDB)
+	backfill, err := backfillStore.LoadBackfill(ctx, backfillID)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to load backfill")
+	}
+	if !backfill.IsTerminalState() {
+		return nil, errors.Wrap(err, "only backfills in terminal states can be retried")
+	}
+	err = backfill.RetryBackfillAttempt(ctx, backfillStore)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to reset backfill")
+	}
+
+	backfillItems, err := backfillStore.GetBackfillQueueInfo(ctx, scheduler.BackfillQueueArgs{ID: &backfill.Id})
+	if err != nil {
+		return nil, err
+	}
+	if len(backfillItems) != 1 {
+		return nil, errors.New("unable to load backfill")
+	}
+	updatedItem := backfillItems[0]
+	return &graphqlbackend.BackfillQueueItemResolver{
+		BackfillID:   updatedItem.ID,
+		InsightTitle: updatedItem.InsightTitle,
+		Label:        updatedItem.SeriesLabel,
+		Query:        updatedItem.SeriesSearchQuery,
+		BackfillStatus: &backfillStatusResolver{
+			queueItem: updatedItem,
+		},
+	}, nil
+}
+func (r *Resolver) MoveInsightSeriesBackfillToFrontOfQueue(ctx context.Context, args *graphqlbackend.BackfillArgs) (*graphqlbackend.BackfillQueueItemResolver, error) {
+	actr := actor.FromContext(ctx)
+	if err := auth.CheckUserIsSiteAdmin(ctx, r.postgresDB, actr.UID); err != nil {
+		return nil, err
+	}
+	var backfillID int
+	err := relay.UnmarshalSpec(args.Id, &backfillID)
+	if err != nil {
+		return nil, errors.Wrap(err, "error unmarshalling the backfill id")
+	}
+	backfillStore := scheduler.NewBackfillStore(r.insightsDB)
+	backfill, err := backfillStore.LoadBackfill(ctx, backfillID)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to load backfill")
+	}
+	if backfill.State != scheduler.BackfillStateProcessing {
+		return nil, errors.Wrap(err, "only backfills ready for precessing can have priority changed")
```

Comment:
```suggestion
		return nil, errors.Wrapf(err, "only backfills ready for processing can have priority changed [current state %v]", backfill.State)
```
not fussed about the wording but communicating state could help